src/live/handleAlbumChat.js

```diff
@@ -1,11 +1,37 @@
-function handleAlbumChat(io, socket, data) {
-  const { message, userId, albumId } = data;
+const ChatMessage = require('../models/ChatMessage');
 
-  // Create a chat room name for the album chat
-  const chatRoomName = `chat:album:${albumId}`;
+async function handleAlbumChat(io, socket, data) {
+  try {
+    const { albumId } = data;
 
-  // Send the message to the room
-  socket.to(chatRoomName).emit('chat:album', { message, userId, albumId });
+    // Create a chat room name for the album chat
+    const chatRoomName = `chat:album:${albumId}`;
+
+    const msg = await ChatMessage.create({ ...data, username: data.user });
+    await msg.populate('user', 'username');
+    console.log('Chat message saved:', msg);
+
+    // Return the _id of the saved message
+    const savedMessage = {
+      ...data,
+      _id: msg._id, // Include _id in the message
+      username: data.username,
```

Comment:
`data` does not have a `username` field based on what the frontend is currently sending us